src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -60,6 +62,67 @@ class DataSource(graphFactory: OrientGraphFactory) extends LazyLogging {
 }
 
 class RawlsTransaction(graph: OrientGraph, dataSource: DataSource) {
+
+  var readLocks:mutable.Map[Element, Int] = mutable.Map[Element, Int]().withDefaultValue(0)
```

Comment:
Sounds legit. The docs say it's the default implementation of `scala.collection.concurrent.Map`, so I'll probably use that.